targets/pna_nic/primitives.cpp

```diff
@@ -0,0 +1,362 @@
+/* Copyright 2024 Marvell Technology, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Rupesh Chiluka (rchiluka@marvell.com)
+ *
+ */
+
+#include <bm/bm_sim/actions.h>
+#include <bm/bm_sim/calculations.h>
+#include <bm/bm_sim/core/primitives.h>
+#include <bm/bm_sim/counters.h>
+#include <bm/bm_sim/meters.h>
+#include <bm/bm_sim/packet.h>
+#include <bm/bm_sim/phv.h>
+#include <bm/bm_sim/logger.h>
+
+#include <random>
+#include <thread>
+
+namespace bm {
+
+namespace pna {
+
+class modify_field : public ActionPrimitive<Data &, const Data &> {
```

Comment:
There are a few legacy primitives here that we kept in simple_switch for historical reasons (they somehow made their way into psa_switch as well), but that may not be relevant to PNA. It would be good to remove what is not needed. I will mark all the ones that I think should be removed with a comment, starting with this one (`modify_field`).